query-results/template.yaml

```diff
@@ -38,6 +38,33 @@ Resources:
       VersioningConfiguration:
         Status: Enabled
 
+  QueryResultsBucketPolicy:
+    Type: AWS::S3::BucketPolicy
+    Properties:
+      Bucket: !Ref QueryResultsBucket
+      PolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Sid: AthenaLambdaAccess
+            Action:
+              - s3:GetBucketLocation
+              - s3:GetObject
+              - s3:ListBucket
+              - s3:ListBucketMultipartUploads
+              - s3:ListMultipartUploadParts
+              - s3:AbortMultipartUpload
+              - s3:PutObject
+            Effect: Allow
+            Resource:
+              - '{{resolve:ssm:QueryResultsBucketArn}}'
+              - '{{resolve:ssm:QueryResultsBucketArn}}/*'
+            Principal:
+              AWS:
+                - '{{resolve:ssm:AuditAccountRootArn}}'
+            Condition:
+              StringLike:
+                aws:PrincipalArn: 'arn:aws:iam::*:role/txma-ticf-integration-InitiateAthenaQueryFunctionR-*'
```

Comment:
If we have declared the account number as an SSM param, you should be able to slap this in here!

Reply:
I don't think we have declared the account number as a param, just the Root Arn.  Could add account number to the config repo?

Reply:
Although considering the Principal is locked to the account anyway, maybe that's a load of extra effort for zero gain!

Reply:
Yep, I think having the principal locked mitigates the risk